rlp/src/stream.rs

```diff
@@ -359,11 +369,11 @@ impl<'a> BasicEncoder<'a> {
 		// 1 byte was already reserved for payload earlier
 		match len {
 			0..=55 => {
-				self.buffer[pos - 1] = 0xc0u8 + len as u8;
+				self.buffer[self.start_pos + pos - 1] = 0xc0u8 + len as u8;
```

Comment:
If `pos` is ever zero it looks like nasty things could happen (was true before too); can you double-check it cannot be called with `pos` == 0?

Reply:
I cannot be sure without auditing all possible encoding paths in this codebase, but this is orthogonal to this PR as it _increases_ the count.

I suggest we merge as is, and add this concern to a round of fuzzing which this library so badly needs anyway.

Reply:
Yepp, agree unrelated to this PR but we should open a ticket for it and investigate it further.